tests/compiler/test_dict.py

```diff
@@ -428,3 +428,56 @@ def lookup(str):
 
     for var, exp_val in zip(vars_to_lookup, expected_values):
         assert lookup(var) == exp_val
+
+
+@pytest.mark.parametrize(
+    "lookup_path, expectation",
+    [
+        ('a["A"]["1"]', True),
+        ('a["A"]["2"]', False),
+        ('a["A"]["3"]', False),
+        ('a["A"]["4"]', False),
+        ('a["A"]', True),
+        ('a["B"]', False),
+        ('a["C"]', False),
+        ('a["D"]', True),
+        ('a["K"]', False),
+        # ('a["K"]["a"]', False),
+        # ('a["K"]["k"]', False),
+        # ('b', False),
+    ],
+)
+def test_dict_is_defined_4317(snippetcompiler, lookup_path, expectation):
+    """
+    Check that calling "is defined" on dictionaries works as syntactic sugar for != null
+    """
+    snippetcompiler.setup_for_snippet(
+        """
+    entity Test:
+        bool value_is_defined
+    end
+
+    implement Test using std::none
+    t = Test()
+
+    a = {
+        "A": {
+            "1": "ok",
+            "2": null,
+            "3": []
+        },
+        "B": null,
+        "C": [],
+        "D": "ok"
```

Comment:
```suggestion
        "D": "ok"
        "E": [null]
```
Tied to the comment above